src/Database/Driver/sparql/Connection.php

```diff
@@ -100,61 +116,43 @@ public function query(string $query, array $args = [], array $options = []): Res
    * {@inheritdoc}
    */
   public function update(string $query, array $args = [], array $options = []): Result {
-    if (!empty($this->logger)) {
-      // @todo Fix this. Logger should have been auto started.
-      // Probably related to the overwritten log object in $this->setLogger.
-      // Look at
-      // \Drupal\webprofiler\StackMiddleware\WebprofilerMiddleware::handle.
-      $this->logger->start('webprofiler');
+    if ($this->logger) {
       $query_start = microtime(TRUE);
     }
 
     try {
-      $results = $this->connection->update($query);
+      // @todo Implement argument replacement. This is not easy as the core
+      // drivers are deferring this operation to \PDOStatement::execute() but
+      // PDO doesn't have any API that exposes the compiled query. We need to
+      // build a routine to perform the arguments SANITIZATION and REPLACEMENT.
+      // @see \PDOStatement::execute()
+      $result = $this->easyRdfClient->update($query);
```

Comment:
OK This explains the missing `$args` that I noted in the test coverage. We should make a followup for this, since it is a feature that can prevent security issues.

Reply:
I think we should raise temporary an `\InvalidArgumentException` if someone is trying to pass arguments, saying that "Passing arguments to query/update is not yet supported."